src/esm_runscripts/filedicts.py

```diff
@@ -341,6 +357,46 @@ def _path_type(self, path: pathlib.Path) -> Union[str, bool]:
         else:
             raise Exception(f"Cannot identify the path's type of {path}")
 
+    def _check_file_info_is_complete(self):
+        error_text = ""
+        missing_vars = ""
+        types_text = ", ".join(self.all_model_filetypes)
+
+        if "type" not in self.keys():
+            error_text = (
+                f"{error_text}"
+                f"- the ``type`` variable is missing. Please define a ``type`` "
+                f"({types_text})\n"
+            )
+            missing_vars = (
+                f"{missing_vars}    ``type``: forcing/input/restart/outdata/...\n"
+            )
+        else:
+            if self["type"] not in self.all_model_filetypes:
+                error_text = (
+                    f"{error_text}"
+                    f"- ``{self['type']}`` is not a supported ``type`` "
+                    f"(``files.{self.name}.type``), please choose one of the following "
+                    f"types: {types_text}\n"
+                )
+                self._original_filedict["type"] = f"``{self._original_filedict['type']}``"
+
+        missing_vars = (
+            f"Please, complete/correct the following vars for your file:\n\n"
+            f"{self.pretty_original_filedict}"
+            f"{missing_vars}"
+        )
+        if error_text:
+            error_text = (
+                f"The file dictionary ``{self.name}`` is missing relevant information "
+                f"or is incorrect:\n{error_text}"
+            )
+            user_error("File Dictionaries", f"{error_text}\n{missing_vars}")
+
+    @property
```

Comment:
Do we really need a `@property` decorator here? It simply prints the output. Not really a getter/setter. Are you planning to add new things to this function?

Reply:
That depends on what we want. This provides you a string representation of the Yaml as a attribute of object. 

Making it a callable would require you capturing that in a variable to use it. 

Also in principle the property decorator does not necessitate additionally defining getter/setter behaviour. It _can_ but doesn't need to. Basically, it is used to have a dot-accessible attribute when the attribute needs some kind of logical computation. 

Maybe this is a global style choice we need to agree upon.

Reply:
I was just trying out a property for the sake of trying. I don't have an opinion on whether it should be a property or not.

Reply:
`@property` removed